tests/integrationv2/test_signature_algorithms.py

```diff
@@ -0,0 +1,176 @@
+    client_options = ProviderOptions(
+        mode=Provider.ClientMode,
+        host="localhost",
+        port=port,
+        cipher=cipher,
+        data_to_send=random_bytes,
+        insecure=True,
+        extra_flags=['-sigalgs', ':'.join(sigs)],
+        protocol=protocol)
+
+    server_options = copy.copy(client_options)
+    server_options.extra_flags = None
+    server_options.data_to_send = None
+    server_options.mode = Provider.ServerMode
+    server_options.key = certificate.key
+    server_options.cert = certificate.cert
+
+    server = managed_process(S2N, server_options, timeout=5)
+    client = managed_process(provider, client_options, timeout=5)
+
+    for results in client.get_results():
+        assert results.exception is None
+        assert results.exit_code == 0
+        assert bytes('Peer signing digest: {}'.format(signature.sig_digest).encode('utf-8')) in results.stdout
+        assert bytes('Peer signature type: {}'.format(signature.sig_type).encode('utf-8')) in results.stdout
+
+    expected_version = get_expected_s2n_version(protocol, provider)
+
+    for results in server.get_results():
+        assert results.exception is None
+        assert results.exit_code == 0
+        assert bytes("Actual protocol version: {}".format(expected_version).encode('utf-8')) in results.stdout
+        assert random_bytes in results.stdout
+
+
+@pytest.mark.uncollect_if(func=skip_ciphers)
+@pytest.mark.parametrize("cipher", ALL_TEST_CIPHERS, ids=get_parameter_name)
+@pytest.mark.parametrize("provider", [OpenSSL])
+@pytest.mark.parametrize("protocol", [Protocols.TLS13, Protocols.TLS12], ids=get_parameter_name)
+@pytest.mark.parametrize("certificate", ALL_TEST_CERTS, ids=get_parameter_name)
+@pytest.mark.parametrize("signature", all_sigs, ids=get_parameter_name)
+@pytest.mark.parametrize("client_auth", [True, False], ids=get_parameter_name)
+def test_s2n_client_signature_algorithms(managed_process, cipher, provider, protocol, certificate, signature, client_auth):
+    port = next(available_ports)
+
+    random_bytes = data_bytes(64)
+    client_options = ProviderOptions(
+        mode=Provider.ClientMode,
+        host="localhost",
+        port=port,
+        cipher=cipher,
+        data_to_send=random_bytes,
+        insecure=True,
+        protocol=protocol)
+
+    server_options = copy.copy(client_options)
+    server_options.extra_flags = None
+    server_options.data_to_send = None
+    server_options.mode = Provider.ServerMode
+    server_options.key = certificate.key
+    server_options.cert = certificate.cert
+    server_options.extra_flags=['-sigalgs', signature.name]
+
+    if client_auth is True:
+        client_options.insecure = False
+        client_options.use_client_auth = True
+        client_options.client_trust_store = Certificates.RSA_2048_SHA256_WILDCARD.cert
+        server_options.key = Certificates.RSA_2048_SHA256_WILDCARD.key
+        server_options.cert = Certificates.RSA_2048_SHA256_WILDCARD.cert
```

Comment:
If we're doing client auth, shouldn't we be setting the key/cert on the client and the trust store on the server?